src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -422,7 +371,44 @@ public static string UrlEncode(string value)
 
         public static byte[] UrlEncodeToBytes(byte[] value, int offset, int count)
         {
-            return UrlEncode(value, offset, count, true /* alwaysCreateNewReturnValue */);
+            if (!ValidateUrlEncodingParameters(value, offset, count))
+            {
+                return null;
+            }
+
+            int cSpaces = 0;
+            int cUnsafe = 0;
+
+            // count them first
+            for (int i = 0; i < count; i++)
+            {
+                char ch = (char)value[offset + i];
+
+                if (ch == ' ')
+                    cSpaces++;
+                else if (!IsUrlSafeChar(ch))
+                    cUnsafe++;
+            }
+
+            // nothing to expand?
+            if (cSpaces == 0 && cUnsafe == 0)
+            {
+                if (offset == 0 && value.Length == count)
```

Comment:
@hughbe No, I just mean that it's unnecessary extra code and can be removed (since the `else` branch should be able to handle all cases)... but it's OK. If you choose not to remove it in this PR, I can submit a new one and cc you.